model/task/task.go

```diff
@@ -994,16 +999,33 @@ func (t *Task) SetAborted(reason AbortInfo) error {
 	)
 }
 
-func (t *Task) SetHasCedarResults(hasCedarResults bool) error {
+// SetHasCedarResults sets the HasCedarResults field of the task to
+// hasCedarResults and, if failedResults is true, sets CedarResultsFailed to
+// true. An error is returned if hasCedarResults is false and failedResults is
+// true as this is an invalid state. Note that if failedResults is false,
+// nothing is set. This is because in cases where separate calls to attach test
```

Comment:
nit: "nothing" is set was initially confusing, but you specifically meant we don't set CedarResultsFailed